### DISCUSSION TITLE
Solr/Zookeeper CVE not recognized - why?

### DISCUSSION THREAD

**Discussion by 3XC1T3D** [aquasecurity/trivy#7954] [Q&A]
### Question

Hi,

we are wondering why following Apache SOLR CVE where not recognized:

- **NVD Link:** https://nvd.nist.gov/vuln/detail/CVE-2024-45217 
  **Apache SOLR Statement:** https://solr.apache.org/security.html#cve-2024-45217-apache-solr-configsets-created-during-a-backup-restore-command-are-trusted-implicitly 

- **NVD Link:** https://nvd.nist.gov/vuln/detail/CVE-2024-51504
  **Zookeeper Statement:** https://zookeeper.apache.org/security.html#CVE-2024-51504

Is there a possibility to configure trivy, that such cve where also recognized?

We use the trivy Docker Image.

Best Regards

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

AlmaLinux 9

### Version

```bash
trivy version
Version: 0.57.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-11-19 06:13:02.20916498 +0000 UTC
  NextUpdate: 2024-11-20 06:13:02.2091647 +0000 UTC
  DownloadedAt: 2024-11-19 09:07:51.10523027 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2024-11-19 02:51:20.64737438 +0000 UTC
  NextUpdate: 2024-11-22 02:51:20.647374259 +0000 UTC
  DownloadedAt: 2024-11-19 09:08:17.049589191 +0000 UTC
```

**Comment by DmitriyLewen** [comment]
Hello @3XC1T3D 

[CVE-2024-45217](https://github.com/advisories/GHSA-h7w9-c5vx-x7j3) should be detected.
Can you share example of your case?

We use `GitHub advisory database` for java advisories.
But GH doesn't have package/fixed version for [CVE-2024-51504](https://github.com/advisories/GHSA-g93m-8x6h-g5gv).
That is why Trivy doesn't detect this CVE.

**Comment by 3XC1T3D** [comment]
Hey @DmitriyLewen ,

because of the CVE-2024-45217 just run following command: `trivy image solr:8.11.3 -f json -o results.json` and you will not find any entry about it.

due to the CVE-2024-51504 before we check our Images, we build our own trivy image like this:
```
FROM aquasec/trivy:latest

ENV TRIVY_DB_REPOSITORY=ghcr.io/aquasecurity/trivy-db,public.ecr.aws/aquasecurity/trivy-db
ENV TRIVY_JAVA_DB_REPOSITORY=ghcr.io/aquasecurity/trivy-java-db,public.ecr.aws/aquasecurity/trivy-java-db

# update baseimage
RUN apk update \
 && apk upgrade \
 && apk add --upgrade bash \
 && apk add curl \
 && rm -rf /var/cache/apk/*

# Fill Trivy caches
RUN trivy image --download-db-only
RUN trivy image --download-java-db-only


COPY certificates/ca.crt /usr/local/share/ca-certificates/ca-certificates.crt
RUN update-ca-certificates

# overwrite entrypoint to be compatible with pipeline
ENTRYPOINT [""]
```

maybe there is the issue?

**Reply by DmitriyLewen** [reply]
> due to the https://github.com/advisories/GHSA-g93m-8x6h-g5gv before we check our Images, we build our own trivy image like this:

No, problem is related with trivy-db.
As i wrote before - for trivy-db we take java advisories from `GitHub Advisory database` - https://trivy.dev/v0.57/docs/scanner/vulnerability/#data-sources_1

GitHub DB doesn't have `package` for this [CVE](https://github.com/advisories/GHSA-g93m-8x6h-g5gv):
![изображение](https://github.com/user-attachments/assets/4e03122a-9c5e-43df-91ae-925a98acb6d6)

Without this field we can't match packages with this CVE.
 
> trivy image solr:8.11.3 -f json -o results.json and you will not find any entry about it.

I checked this and a bit confusing:
1. Trivy correctly detect `solr-*` packages:
   ```bash
   ➜ trivy -q image solr:8.11.3 -f json --list-all-pkgs | grep ' "Name": "org.apache.solr:'
          "Name": "org.apache.solr:solr-analysis-extras",
          "Name": "org.apache.solr:solr-analytics",
          "Name": "org.apache.solr:solr-cell",
          "Name": "org.apache.solr:solr-core",
          "Name": "org.apache.solr:solr-core",
          "Name": "org.apache.solr:solr-dataimporthandler",
          "Name": "org.apache.solr:solr-dataimporthandler-extras",
          "Name": "org.apache.solr:solr-gcs-repository",
          "Name": "org.apache.solr:solr-jaegertracer-configurator",
          "Name": "org.apache.solr:solr-langid",
          "Name": "org.apache.solr:solr-ltr",
          "Name": "org.apache.solr:solr-prometheus-exporter",
          "Name": "org.apache.solr:solr-s3-repository",
          "Name": "org.apache.solr:solr-solrj",
          "Name": "org.apache.solr:solr-solrj",
          "Name": "org.apache.solr:solr-test-framework",
          "Name": "org.apache.solr:solr-velocity",
   ```
   But `solr:8.11.3` doesn't have jar file for `solr` package (e.g. `solr-8.11.3.jar`):
   ```bash
   solr@4797102414e1:/opt/solr-8.11.3$ find / -name "solr*.*ar"
    find: ‘/etc/ssl/private’: Permission denied
    find: ‘/proc/tty/driver’: Permission denied
    /opt/solr-8.11.3/server/solr-webapp/webapp/WEB-INF/lib/solr-solrj-8.11.3.jar
    /opt/solr-8.11.3/server/solr-webapp/webapp/WEB-INF/lib/solr-core-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-solrj-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-gcs-repository-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-dataimporthandler-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-velocity-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-analysis-extras-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-langid-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-jaegertracer-configurator-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-prometheus-exporter-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-core-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-cell-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-dataimporthandler-extras-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-test-framework-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-ltr-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-analytics-8.11.3.jar
    /opt/solr-8.11.3/dist/solr-s3-repository-8.11.3.jar
    find: ‘/root’: Permission denied
    find: ‘/var/cache/ldconfig’: Permission denied
    find: ‘/var/cache/apt/archives/partial’: Permission denied
   ```
2. I thought that perhaps `solr` uses another file name, but last version for `solr` in maven central is `4.10.4` - https://mvnrepository.com/artifact/org.apache.solr/solr

**Reply by 3XC1T3D** [reply]
> No, problem is related with trivy-db. As i wrote before - for trivy-db we take java advisories from `GitHub Advisory database` - https://trivy.dev/v0.57/docs/scanner/vulnerability/#data-sources_1
> 
> GitHub DB doesn't have `package` for this [CVE](https://github.com/advisories/GHSA-g93m-8x6h-g5gv): ![изображение](https://private-user-images.githubusercontent.com/91113035/387946034-4e03122a-9c5e-43df-91ae-925a98acb6d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwOTAxNzksIm5iZiI6MTczMjA4OTg3OSwicGF0aCI6Ii85MTExMzAzNS8zODc5NDYwMzQtNGUwMzEyMmEtOWM1ZS00M2RmLTkxYWUtOTI1YTk4YWNiNmQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDA4MDQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkZDM1MzA5M2QzYjQzY2EyN2U5OTBkNzc0MTc5MmIyMDc4NjZiM2VhNjgxN2JjMTQyYjZmNmU5NzQ2MzA4YzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._Gco0ie2o7Eyubw8tsTVctA7CtF8k6wpwj3f5sD9s08)
> 
> Without this field we can't match packages with this CVE.

How you configure the GitHub Advisory database for the java advisories? or are these the default settings?

> > trivy image solr:8.11.3 -f json -o results.json and you will not find any entry about it.
> 
> I checked this and a bit confusing:
> 
> 1. Trivy correctly detect `solr-*` packages:
>    ```shell
>    ➜ trivy -q image solr:8.11.3 -f json --list-all-pkgs | grep ' "Name": "org.apache.solr:'
>           "Name": "org.apache.solr:solr-analysis-extras",
>           "Name": "org.apache.solr:solr-analytics",
>           "Name": "org.apache.solr:solr-cell",
>           "Name": "org.apache.solr:solr-core",
>           "Name": "org.apache.solr:solr-core",
>           "Name": "org.apache.solr:solr-dataimporthandler",
>           "Name": "org.apache.solr:solr-dataimporthandler-extras",
>           "Name": "org.apache.solr:solr-gcs-repository",
>           "Name": "org.apache.solr:solr-jaegertracer-configurator",
>           "Name": "org.apache.solr:solr-langid",
>           "Name": "org.apache.solr:solr-ltr",
>           "Name": "org.apache.solr:solr-prometheus-exporter",
>           "Name": "org.apache.solr:solr-s3-repository",
>           "Name": "org.apache.solr:solr-solrj",
>           "Name": "org.apache.solr:solr-solrj",
>           "Name": "org.apache.solr:solr-test-framework",
>           "Name": "org.apache.solr:solr-velocity",
>    ```     
>    But `solr:8.11.3` doesn't have jar file for `solr` package (e.g. `solr-8.11.3.jar`):
>    ```shell
>    solr@4797102414e1:/opt/solr-8.11.3$ find / -name "solr*.*ar"
>     find: ‘/etc/ssl/private’: Permission denied
>     find: ‘/proc/tty/driver’: Permission denied
>     /opt/solr-8.11.3/server/solr-webapp/webapp/WEB-INF/lib/solr-solrj-8.11.3.jar
>     /opt/solr-8.11.3/server/solr-webapp/webapp/WEB-INF/lib/solr-core-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-solrj-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-gcs-repository-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-dataimporthandler-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-velocity-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-analysis-extras-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-langid-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-jaegertracer-configurator-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-prometheus-exporter-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-core-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-cell-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-dataimporthandler-extras-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-test-framework-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-ltr-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-analytics-8.11.3.jar
>     /opt/solr-8.11.3/dist/solr-s3-repository-8.11.3.jar
>     find: ‘/root’: Permission denied
>     find: ‘/var/cache/ldconfig’: Permission denied
>     find: ‘/var/cache/apt/archives/partial’: Permission denied
>    ```
> 2. I thought that perhaps `solr` uses another file name, but last version for `solr` in maven central is `4.10.4` - https://mvnrepository.com/artifact/org.apache.solr/solr

the central solr file is named solr-core-8.11.3.jar :see_no_evil: - https://mvnrepository.com/artifact/org.apache.solr/solr-core

**Reply by DmitriyLewen** [reply]
> How you configure the GitHub Advisory database for the java advisories? or are these the default settings?

GitHub uses `OSV` format. So we just read advisories from https://github.com/github/advisory-database and store into `trivy-db` (https://github.com/aquasecurity/trivy-db/blob/main/pkg/vulnsrc/ghsa/ghsa.go)

> the central solr file is named solr-core-8.11.3.jar 🙈 - https://mvnrepository.com/artifact/org.apache.solr/solr-core

hm... there is no information in the documents related to CVE-2024-45217 that the solr-core package is affected.
If you can confirm that `solr-core` is vulnerable - can you suggest changing package name in GitHub? (https://github.com/advisories/GHSA-h7w9-c5vx-x7j3/improve)

BTW maven central also doesn't mark `8.11.3` as vulnerable:
![screenshot](https://github.com/user-attachments/assets/5042f276-ca25-4c7e-a4ed-9ef104a7b32b)

**Reply by 3XC1T3D** [reply]
> GitHub uses `OSV` format. So we just read advisories from https://github.com/github/advisory-database and store into `trivy-db` (https://github.com/aquasecurity/trivy-db/blob/main/pkg/vulnsrc/ghsa/ghsa.go)

sorry i dont get it - could you please explain (maybe step by step) how you do this with the trivy docker image or a native trivy installation? thank you

> hm... there is no information in the documents related to [CVE-2024-45217](https://github.com/advisories/GHSA-h7w9-c5vx-x7j3) that the solr-core package is affected. If you can confirm that `solr-core` is vulnerable - can you suggest changing package name in GitHub? (https://github.com/advisories/GHSA-h7w9-c5vx-x7j3/improve)

no i cant confirm if solr-core is vulnerable , we are just wondering why both cve were not deteced from trivy :/

**Reply by knqyf263** [reply]
>sorry i dont get it - could you please explain (maybe step by step) how you do this with the trivy docker image or a native trivy installation? thank you

We build the vulnerability database using https://github.com/advisories?query=type%3Areviewed+ecosystem%3Amaven. That's it.

**Reply by 3XC1T3D** [reply]
> >sorry i dont get it - could you please explain (maybe step by step) how you do this with the trivy docker image or a native trivy installation? thank you
> 
> We build the vulnerability database using https://github.com/advisories?query=type%3Areviewed+ecosystem%3Amaven. That's it.

Oh man.... know i get it 🙈 i didnt understand that when you say 'we' you intended to say 'trivy'...... Sry for the misunderstanding...

**Reply by knqyf263** [reply]
Oh, you didn't know we're maintainers. Yes, 'we' means 'Trivy'. The Trivy project builds the [database](https://github.com/aquasecurity/trivy-db) on GitHub Actions and downloads it during scanning.
